### DISCUSSION TITLE
Can Supabase's Real-Time Sync and Vector Features Work with Direct Inserts via PostgreSQL ORM?

### DISCUSSION THREAD

**Discussion by jais001** [supabase/supabase#30588] [Questions] [answered]
Hi,
I’m exploring Supabase’s features, especially real-time synchronization and its vector database capabilities.

My question is:

If I insert data (including vector embeddings) directly into the Supabase database using a PostgreSQL ORM (e.g., sqlalchemy) instead of the Supabase client python SDK:

Will these inserts still trigger real-time synchronization updates for subscribed clients?
Can I still take advantage of Supabase’s vector database features, such as similarity search and efficient indexing?
Are there any specific configurations or limitations I should be aware of when using external ORMs for interacting with the Supabase database?

Will other Supabase features, such as Edge Functions and Row-Level Security (RLS), remain functional and synchronized if data is manipulated outside the Supabase SDK?

Thanks in advance...

**Comment by GaryAustin1** [comment, marked as answer]
Realtime monitors Postgres database changes directly and does not care what method the database is changed.  If you are using RLS then your clients would need to meet RLS on the tables.

The other vector features you mention are all part of Postgres and do not depend on any Supabase SDKs.

RLS can be tricky from a direct database connection because you normally run as Postgres and RLS for the Supabase clients normally depend on decoding the JWT of the user for information.  There are ways you can change the Postgres role and set database settings simlar to what the REST database client does if you need to meet the same policies.
